powershell-yaml.psm1

```diff
@@ -38,19 +38,43 @@ function Convert-ValueToProperType {
         if (!($Value -is [string])) {
             return $Value
         }
-        $culture = New-Object System.Globalization.CultureInfo("invariant")
+        $culture = 
         $types = @([int], [long], [double], [boolean], [decimal])
         foreach($i in $types){
-            try {
-                if ($i.IsAssignableFrom([boolean])){
-                    return $i::Parse($Value)
-                } else {
-                    return $i::Parse($Value, $culture)
+            $parsedValue = New-Object -TypeName $i.FullName
+            if ($i.IsAssignableFrom([boolean])){
+                $result = $i::TryParse($Value,[ref]$parsedValue) 
+            } else {
+                $result = $i::TryParse($Value, [Globalization.NumberStyles]::Any, [Globalization.CultureInfo]::CurrentCulture, [ref]$parsedValue)
```

Comment:
```TryParse``` should work with any culture. There is no reason why it should fail, although, results will vary for some types like ```[decimal]```. See above comment and issue.